src/services/factories/tokenList.ts

```diff
@@ -175,18 +176,56 @@ export function getTokensFactory(factoryParams: {
     const tokenDetailsPromises: (Promise<TokenDetails | undefined> | TokenDetails)[] = []
     addressToIdMap.forEach((id, tokenAddress) => {
       // Resolve the details using the config, otherwise fetch the token
-      const token: TokenDetails | undefined | Promise<TokenDetails | undefined> = tokensConfigMap.has(tokenAddress)
-        ? tokensConfigMap.get(tokenAddress)
+      const token: undefined | Promise<TokenDetails | undefined> = tokensConfigMap.has(tokenAddress)
```

Comment:
nit: can we in the future call unresolved vars `promised<varName>`? I find it easier to read.